### PR TITLE
Add V2 primitive Q-Ctrl tests 

### LINKED ISSUE #1499: Write integration tests for v2 primitives for Q-Ctrl
**What is the expected feature or enhancement?**

V2 primitives are currently disabled if `channel_strategy == "q-ctrl"` is specified, even though Q-Ctrl does support V2. There should be some integration tests to ensure they work properly, and then they can be enabled.

**Acceptance criteria**

- Have some basic integration tests for Q-Ctrl
- Allow V2 primitives to be called with Q-Ctrl channel strategy, assuming there are no major issues found by the tests

### PR DESCRIPTION
**Pull request by kt474**
### Summary

Ready for initial review - wasn't completely sure how detailed the tests should be, my understanding is that we're just testing "does it run"/happy path.

### Details and comments
Fixes #1499

### CODE REVIEW

File: test/qctrl/test_qctrl.py [hunk 1]
```diff
@@ -28,6 +39,91 @@
 DIFFERENCE_THRESHOLD = 0.35
 
 
+class TestV2PrimitivesQCTRL(IBMIntegrationTestCase):
+    """Integration tests for V2 primitives using QCTRL."""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.bell = bell()
+        self.backend = "alt_canberra"
```
Comment (jyu00) [inline test/qctrl/test_qctrl.py:48]: We shouldn't hard code backend name

File: test/qctrl/test_qctrl.py [hunk 2]
```diff
@@ -28,6 +39,91 @@
...
+
+    @run_integration_test
+    def test_sampler_v2_qctrl(self, service):
+        """Test qctrl bell state with samplerV2"""
+        shots = 1000
```
Comment (jyu00) [inline test/qctrl/test_qctrl.py:53]: Let's save some QPU time if we are not verifying result data

```suggestion
        shots = 1
```

**Review by jyu00 — COMMENTED** [review]
#1499 also says

`Allow V2 primitives to be called with Q-Ctrl channel strategy, assuming there are no major issues found by the tests`

**Comment by kt474** [comment]
> #1499 also says
> 
> `Allow V2 primitives to be called with Q-Ctrl channel strategy, assuming there are no major issues found by the tests`

It was enabled in https://github.com/Qiskit/qiskit-ibm-runtime/pull/1550

File: test/qctrl/test_qctrl.py [hunk 3]
```diff
@@ -28,6 +39,90 @@
...
+        self.backend = self.service.least_busy(simulator=False)
+
+    @run_integration_test
+    def test_sampler_v2_qctrl(self, service):
+        """Test qctrl bell state with samplerV2"""
+        shots = 1
+        bell_circuit = QuantumCircuit(2)
+        bell_circuit.h(0)
+        bell_circuit.cx(0, 1)
+        bell_circuit.measure_all()
```
Comment (jyu00) [inline test/qctrl/test_qctrl.py:57]: nit - why create your own bell circuit when there is a self.bell?

File: test/qctrl/test_qctrl.py [hunk 4]
```diff
@@ -28,6 +39,90 @@
...
+        with Session(service, backend=self.backend):
+            sampler = SamplerV2()
+
+            result = sampler.run([isa_circuit], shots=shots).result()
+            self._verify_sampler_result(result, num_pubs=1)
+
+    @run_integration_test
+    def test_estimator_v2_qctrl(self, service):
+        """Test simple circuit with estimatorV2 using qctrl."""
+        backend = service.backend(self.backend)
```
Comment (jyu00) [inline test/qctrl/test_qctrl.py:71]: I don't think this would work because `self.backend` is already a backend object.

Comment (kt474) [reply test/qctrl/test_qctrl.py]: Oops, missed that - all tests passing now

**Pull request merged**